design/FY2018/AirSystemSizingAdjust_Std62_1_refactor.md

```diff
@@ -0,0 +1,97 @@
+
+SimAirServingZones.cc, modify routine UpdateSysSizing() - Move Standard 62.1 variables out to DataSizing.hh. Store state of new booleans sysSizeCoolingDominant and sysSizeHeatingDominant in FinalSysSizing data structure to know how the air system main design flow was determined. Remove summary table reporting, to be replaced by new cleaner code in new routine ManageSystemVentilationAdjustments in SizingManager.cc 
+
+DataSizing.cc/.hh add data variables for Standard 62.1 calculations, move variables buried in SimAirServingZones.cc subroutines and put into DataSizing.cc where the variables persist longer and can be accessed. 
+
+SizingManager.cc, new routine ManageSystemSizingAdjustments() - This routine adjusts system sizing outcomes based on how the zone air terminals finish out their sizing. The zone models are executed to trigger their sizing routines. Then the air terminal units data structures are scanned to sum design flow rates. Every air terminal connected to a particular air loop is summed for: 1) minimum heating flow rate, 2) maximum heating flow rate, 3) maximum flow rate, and 4) store zone level flow information for Standard 62.1 calculations, Vpz, Vpz_min, Vdz, and Vdz_min for both cooling and heating. The summed values are used to "Adjust" the system sizing results.  The corrected values are used to autosize the central heating flow ratio, if set to autosize by the user.  
+
+SizingManager.cc, new routine ManageSystemVentilationAdjustments() Redo Standard 62.1 VRP calculations using latest information on zone flows and report to tables.  Collect all summary table reporting one place with organized code.  
+
+SizingManager.cc, new routine DetermineSystemPopulationDiversity(). New routine to determine Pz sum, Ps, and D for each air system for Standard 62.1 calculations. Temporarily stop with system DoingSizing and do a side calculation that marches through all timesteps for each runperiod and sum up the concurrent max occupants in all the zones by air system.  If the input file has no RunPeriod objects, then use the design days.  This factors in all the scheduled detail for an accurate calculation of system population diversity, D.  Current code uses the schedule max for each people object but that cannot handle the coincident air system total unless all the schedules in the building have maximums at the same time of day. 
```

Comment:
@EnergyArchmage In ScheduleManager there are functions for ScheduleAnnualFullLoadHours and ScheduleHoursGT1perc which scan through a full year for a given schedule.  There just walk through weekdays, and appear to ignore special days - I don't know if that's required here but it could be added.  Seems like a similar approach could be used here.

Reply:
I mean these walk through the days of the week, including weekends.